mpas_analysis/ocean/meridional_overturning_circulation.py

```diff
@@ -483,18 +463,18 @@ def write_file(dsMOCTimeSeries):
             first = False
         date = days_to_datetime(ds.Time[tIndex], calendar=calendar)
         print '     date: {:04d}-{:02d}'.format(date.year, date.month)
-        horizontalVel = ds.avgNormalVelocity[tIndex, :, :].values
-        verticalVel = ds.avgVertVelocityTop[tIndex, :, :].values
-        velArea = verticalVel * areaCell[:, np.newaxis]
-        transportZ = _compute_transport(maxEdgesInTransect,
-                                        transectEdgeGlobalIDs,
-                                        transectEdgeMaskSigns,
+        # Cache results if their size is reasonable
+        load = partial(preload, maxgb=config.getfloat('input',
+                                                      'reasonableArraySizeGB'))
+        horizontalVel = load(ds.avgNormalVelocity.isel(Time=tIndex))
+        velArea = load(ds.avgVertVelocityTop.isel(Time=tIndex) * areaCell)
```

Comment:
It's kind of a style choice but I'd prefer having it be as straightforward as possible for someone who might use this code as a starting point to figure out how to do something similar in another analysis.